src/Components/Server/src/Circuits/RemoteNavigationManager.cs

```diff
@@ -65,9 +67,14 @@ public void NotifyLocationChanged(string uri, string state, bool intercepted)
         NotifyLocationChanged(intercepted);
     }
 
+    public ValueTask<bool> HandleLocationChanging(string uri, bool intercepted, bool forceLoad)
+    {
+        return NotifyLocationChanging(uri, intercepted, forceLoad);
+    }
+
     /// <inheritdoc />
     [DynamicDependency(DynamicallyAccessedMemberTypes.PublicProperties, typeof(NavigationOptions))]
-    protected override void NavigateToCore(string uri, NavigationOptions options)
+    protected override async void NavigateToCore(string uri, NavigationOptions options)
```

Comment:
`async void` is going to raise some eyebrows.

At the very least, this method ensure errors are handled (e.g., `try`/`catch` everything, and feed exceptions back into the system somehow).

Possibly better might be keeping `NavigateToCore` as `void`, but only calling it *after* we've been through some new optional async `BeforeNavigation` flow.